### PR TITLE
add nationalcode validate

### PR DESCRIPTION
**Pull request by PouryaMansouri**

### CODE REVIEW

File: nakhll_market/serializers.py [hunk 1]
```diff
@@ -645,6 +645,15 @@ class Meta:
...
         }
 
+    def validate_NationalCode(self, value):
+        if not value:
+            raise serializers.ValidationError(
+                'وارد کردن کد ملی الزامی است'
+            )
+        elif len(value) != 10:
+            raise serializers.ValidationError('کد ملی باید 10 رقم باشد')
+        return value
```
Comment (likecodingloveproblems) [inline nakhll_market/serializers.py:655]: national code must be digit only!
```suggestion
        elif value.isdigit():
            raise serializers.ValidationError('کد ملی باید رقم باشد')
        return value
```

Comment (milad7212) [reply nakhll_market/serializers.py:655]: now it is just a number

File: nakhll_market/serializers.py [hunk 2]
```diff
@@ -654,6 +663,7 @@ class Meta:
         fields = ['User_Profile']
 
 
+
```
Comment (likecodingloveproblems) [inline nakhll_market/serializers.py:666]: @PouryaMansouri 
it seems styling only adds 2 lines between classes, am I right?

File: nakhll_market/serializers.py (hunk 1 above)
Comment (PouryaMansouri) [reply nakhll_market/serializers.py:655]: > national code must be digit only!

It's better to store national code on the string to check it with regex later.

Comment (PouryaMansouri) [reply nakhll_market/serializers.py:655]: and this was only the first step of validation.
it should be completed later.

File: nakhll_market/serializers.py (hunk 2 above)
Comment (PouryaMansouri) [reply nakhll_market/serializers.py:666]: yes.
for clean code structure.

File: nakhll_market/serializers.py (hunk 1 above)
Comment (likecodingloveproblems) [reply nakhll_market/serializers.py:655]: > > national code must be digit only!
> 
> It's better to store national code on the string to check it with regex later.

Ok, isdigit only checks that string is numbers only or not, it's a business rule, you know we don't have alphabetic chars in national code in the real world.

Comment (likecodingloveproblems) [reply nakhll_market/serializers.py:655]: > and this was only the first step of validation. it should be completed later.

I don't get the point, what do you mean?
Do you want to improve validation later?
Do you know dirty data can make enhancements painful ?!

Comment (PouryaMansouri) [reply nakhll_market/serializers.py:655]: > > and this was only the first step of validation. it should be completed later.
> 
> I don't get the point, what do you mean? Do you want to improve validation later? Do you know dirty data can make enhancements painful ?!

I am going to add the validation of the national code tomorrow.

**Comment by likecodingloveproblems** [comment]
@PouryaMansouri why do you close and delete this PR?

**Comment by PouryaMansouri** [comment]
> @PouryaMansouri why do you close and delete this PR?

I opened a new issue for national code validation ( and I want to add completed  the validation on that issue)

**Pull request closed without merging**